tools/check_code_style.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
```

Comment:
Missing license header.  Must be included on all scripts.  Should this be apache or BSD.  BSD is safe for now because all of the other files have BSD headers and this will be properly updated to the apache license when that effort begins.  It cannot come into the repository with no license.

Reply:
I was looking at this as in incremental steps with reusable pieces that we are free to evolve. 
The real value add in this PR is the ability to fix in ones editor the CS violations.

NuttX has no complete CS solution. It is impossible to enforce a CS without tools that work 100% and is not go-no-go. for CI and pr-commit hooks.

We should not expect professional developers and new members of the community to have to read 10 emails on how each one of the current tools do not work and have them understand that the indent.sh will destroy a formatted file. What will help is a proper tool like Astyle, Uncurstify, clang-format that can detect and FIX the code allong with a few gold standard files (h,c,Kconfig) to validate it with.

At this point. I think there are there options.

1. We fix the tooling with all your suggestions.
2. We remove the tooling commit (I keep them separate on purpose) until we have the work flow nailed.